modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.module

```diff
@@ -294,32 +269,32 @@ function social_group_flexible_group_group_access(EntityInterface $entity, $oper
   }
 
   // If group visibility value doesn't exist we can skip.
-  if (empty($entity->getFieldValue('field_flexible_group_visibility', 'value'))) {
+  if ($entity->field_flexible_group_visibility->isEmpty()) {
     return $result;
   }
 
-  // If group visibility exists and public is selected, we can skip.
-  $group_visibility = $entity->getFieldValue('field_flexible_group_visibility', 'value');
-  if ($group_visibility === 'public') {
-    return $result;
-  }
+  switch ($entity->field_flexible_group_visibility->value) {
```

Comment:
Because the "getString" will broke functionality when we decide to make this field multiple. Also, when the "getValue" method will be used then the code be longer so since Drupal 8 provides magic properties so I'd prefer to use this modern style and don't back to the code like in Drupal 7.